---------------------
PatchSet 2805 
Date: 2012/11/18 19:51:52
Author: bfriesen
Branch: HEAD
Tag: (none) 
Log:
* libtiff/tif_{unix,vms,win32}.c (_TIFFmalloc): ANSI C does not
require malloc() to return NULL pointer if requested allocation
size is zero.  Assure that _TIFFmalloc does.

Members: 
	ChangeLog:1.924->1.925 
	libtiff/tif_unix.c:1.23->1.24 
	libtiff/tif_vms.c:1.11->1.12 
	libtiff/tif_win32.c:1.39->1.40 

Index: libtiff/libtiff/tif_unix.c
diff -u libtiff/libtiff/tif_unix.c:1.23 libtiff/libtiff/tif_unix.c:1.24
--- libtiff/libtiff/tif_unix.c:1.23	Fri Jun  1 16:40:59 2012
+++ libtiff/libtiff/tif_unix.c	Sun Nov 18 12:51:52 2012
@@ -257,6 +257,9 @@
 void*
 _TIFFmalloc(tmsize_t s)
 {
+        if (s == 0)
+                return ((void *) NULL);
+
 	return (malloc((size_t) s));
 }
 

